app/controllers/api/vms_controller.rb

```diff
@@ -322,55 +323,55 @@ def validate_vm_for_remote_console(vm, protocol = nil)
 
     def start_vm(vm)
       desc = "#{vm_ident(vm)} starting"
-      task_id = queue_object_action(vm, desc, :method_name => "start", :role => "ems_operations")
+      task_id = queue_object_action(vm, desc, queue_options("start"))
       action_result(true, desc, :task_id => task_id)
     rescue => err
       action_result(false, err.to_s)
     end
 
     def stop_vm(vm)
       desc = "#{vm_ident(vm)} stopping"
-      task_id = queue_object_action(vm, desc, :method_name => "stop", :role => "ems_operations")
+      task_id = queue_object_action(vm, desc, queue_options("stop"))
       action_result(true, desc, :task_id => task_id)
     rescue => err
       action_result(false, err.to_s)
     end
 
     def suspend_vm(vm)
       desc = "#{vm_ident(vm)} suspending"
-      task_id = queue_object_action(vm, desc, :method_name => "suspend", :role => "ems_operations")
+      task_id = queue_object_action(vm, desc, queue_options("suspend"))
       action_result(true, desc, :task_id => task_id)
     rescue => err
       action_result(false, err.to_s)
     end
 
     def pause_vm(vm)
       desc = "#{vm_ident(vm)} pausing"
-      task_id = queue_object_action(vm, desc, :method_name => "pause", :role => "ems_operations")
+      task_id = queue_object_action(vm, desc, queue_options("pause"))
       action_result(true, desc, :task_id => task_id)
     rescue => err
       action_result(false, err.to_s)
     end
 
     def shelve_vm(vm)
       desc = "#{vm_ident(vm)} shelving"
-      task_id = queue_object_action(vm, desc, :method_name => "shelve", :role => "ems_operations")
+      task_id = queue_object_action(vm, desc, queue_options("shelve"))
       action_result(true, desc, :task_id => task_id)
     rescue => err
       action_result(false, err.to_s)
     end
 
     def shelve_offload_vm(vm)
       desc = "#{vm_ident(vm)} shelve-offloading"
-      task_id = queue_object_action(vm, desc, :method_name => "shelve_offload", :role => "ems_operations")
+      task_id = queue_object_action(vm, desc, queue_options("shelve_offload"))
       action_result(true, desc, :task_id => task_id)
     rescue => err
       action_result(false, err.to_s)
     end
 
     def destroy_vm(vm)
       desc = "#{vm_ident(vm)} deleting"
-      task_id = queue_object_action(vm, desc, :method_name => "destroy")
```

Comment:
hmm, was the :role missing here by mistake ? we're now introducing it with queue_options.